packages/sirius-web/frontend/sirius-web-application/src/modals/upload-document/UploadDocumentModal.tsx

```diff
@@ -145,6 +175,25 @@ export const UploadDocumentModal = ({ editingContextId, onDocumentUploaded, onCl
             data-testid="upload-document-submit">
             Upload
           </Button>
+          <Button
+            variant="contained"
+            disabled={uploadDocumentModal !== 'success'}
+            color="primary"
+            type="button"
+            form="upload-form-id"
+            data-testid="upload-document-download-report"
+            onClick={() => onDownloadReport()}>
+            Download Report
+          </Button>
```

Comment:
Could you move this download action in a div with the downloadReportMessage above and ensure that it is only displayed if there is a report message. Right now it is activated and I can click on it even if there are no report which is a bit odd. Thanks